exploration/0474-text-vs-code.md

```diff
@@ -103,7 +130,49 @@ equivalent of `\n`.
 
 ## Proposed Design
 
-TBD
+**Start in text mode**
+
+In this option, whitespace is trimmed from `pattern` constructs unless
+the pattern is quoted.
+
+```
+Hello world!
+
+Hello {$user}!
+
+{input $now :datetime dateStyle=long}
+Hello {$user}. Today is {$now}
+
+{local $now = {:systemGetCurrentTime :datetime dateStyle=medium}}
+Hello {$user}. Today is {$now}
+
+{match {$count :number integer=true}}
+{when 0}   Hello {$user}. Today is {$now} and you have no geese.
+{when one} Hello {$user}. Today is {$now} and you have {$count} goose.
+{when few} {  Hello {$user}, this message has spaces on the front and end.  }
+{when *}   Hello {$user}. Today is {$now} and you have {$count} geese.
+```
+
+Bear in mind that whitespace has no meaning in our syntax,
+so some of the above messages are actually:
+
+```
+{input $now :datetime dateStyle=long}Hello {$user}. Today is {$now}
+
+{match {$count :number option=value}}{when 0} Hello {$user}{when one} Hello {$user}{when *}{ Hello {$user} }
+```
+
+Key choices we should consider as a working group:
+
+- Do we like the double-brackets that `local` and `match` require?
```

Comment:
At least for `local`, the inner brackets can be dropped with no loss of comprehension or meaning.

Reply:
I thought about that. Would that be confusing, though? The right hand side is otherwise an expression. There are many ways to write expressions. Do we prefer:

```
{local $foo = {$bar :function option=value}}   // current with expression
{local $foo = $bar :function option=value}     // remove expression syntax
{local $foo $bar :function option=value}       // its own production, maybe hard to grok?
```

I like the middle one the most (less noise). We could make the top one optional.

What about `match`?

Reply:
For `match`, I'd really like it if at least
```
{match $foo}
```
was recognised as effectively shorthand for
```
{match {$foo}}
```